src/main/kotlin/org/jetbrains/research/transformations/Anonymization.kt

```diff
@@ -0,0 +1,75 @@
+@file:Suppress("UNCHECKED_CAST")
+
+package org.jetbrains.research.transformations
+
+import com.github.gumtreediff.tree.ITree
+import com.github.gumtreediff.tree.Tree
+import com.github.gumtreediff.tree.TreeContext
+
+object Anonymization: Transformation {
+    override val metadataKey = "anonymization"
+    private var anonymNamesMap: MutableMap<String, String> = mutableMapOf()
+    private var numOfVariables = 0
+    private const val VAR_PREFIX = "v"
+
+
+    override fun apply(treeCtx: TreeContext, toStoreMetadata: Boolean) {
+        anonymNamesMap = mutableMapOf()
+        numOfVariables = 0
+        val tree = treeCtx.root
+        val treeIterator = tree.trees.iterator()
+
+        while (treeIterator.hasNext()) {
+            val node = treeIterator.next()
+
+            if (treeCtx.getTypeLabel(node) == "NameStore") {
```

Comment:
I think you should create an [enum](https://kotlinlang.org/docs/reference/enum-classes.html) class to store node types because you use it a lot of times. What do you think?

And then you should `when` operator for checking the type. See an example [here](https://stackoverflow.com/questions/53138172/how-to-implement-switch-case-statement-in-kotlin).

```
enum class NodeType(val type: String) {
        NAME_STORE("NameStore"),
        NAME_LOAD("NameLoad)
}

val nodeType = NodeType(treeCtx.getTypeLabel(node))
when (nodeType) {
    NAME_STORE -> { ... }
    NAME_LOAD -> { ... }
    else -> { 
        // default actions, for example log a message
    }
}
```